src/WebJobs.Script/Description/CSharp/FunctionEntryPointResolver.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public class FunctionEntryPointResolver : IFunctionEntryPointResolver
+    {
+        public MethodInfo GetFunctionEntryPoint(IList<MethodInfo> declaredMethods)
+        {
+            if (declaredMethods.Count == 1)
+            {
+                return declaredMethods[0];
+            }
+
+            var attributedMethods = new List<MethodInfo>();
+            var runMethods = new List<MethodInfo>();
+
+            foreach (var methodInfo in declaredMethods)
+            {
+                if (methodInfo.GetCustomAttribute<FunctionEntryPointAttribute>() != null)
+                {
+                    attributedMethods.Add(methodInfo);
+                }
+
+                if (string.Compare(methodInfo.Name, "run", true) == 0)
+                {
+                    runMethods.Add(methodInfo);
+                }
+            }
+
+            if (attributedMethods.Count == 1)
+            {
+                return attributedMethods[0];
+            }
+
+            if (attributedMethods.Count > 1)
+            {
+                throw CreateCompilationException("AF002", "Multiple function entry points", $"Only one method may be decorated with the {nameof(FunctionEntryPointAttribute)}.");
+            }
+
+            if (runMethods.Count == 1)
+            {
+                return runMethods[0];
+            }
+
+            if (attributedMethods.Count > 1)
```

Comment:
should this be runMethods.Count > 1? the error message below mentions "methods named run"